readthedocs/rtd_tests/tests/test_unresolver.py

```diff
@@ -0,0 +1,85 @@
+from django.test import override_settings
+import django_dynamic_fixture as fixture
+import pytest
+
+from readthedocs.rtd_tests.tests.test_resolver import ResolverBase
+from readthedocs.core.unresolver import unresolve
+from readthedocs.projects.models import Domain
+
+
+@override_settings(
+    PUBLIC_DOMAIN='readthedocs.io',
+    RTD_EXTERNAL_VERSION_DOMAIN='dev.readthedocs.build',
+)
+@pytest.mark.proxito
+class UnResolverTests(ResolverBase):
+
+    def test_unresolver(self):
+        parts = unresolve('http://pip.readthedocs.io/en/latest/foo.html#fragment')
+        self.assertEqual(parts.project.slug, 'pip')
+        self.assertEqual(parts.language_slug, 'en')
+        self.assertEqual(parts.version_slug, 'latest')
+        self.assertEqual(parts.filename, 'foo.html')
+        self.assertEqual(parts.fragment, 'fragment')
```

Comment:
I love these tests! Super easy to follow! :100:  More "unit test" that our current tests that needs to call Django internals